api/lib/stores/global-store.js

```diff
@@ -2,67 +2,50 @@ const Sentry = require('@sentry/node')
 
 class GlobalStore {
   constructor(database) {
-    this.stores = []
     this.db = database
+    this.store = {}
     this.globalValidators = {}
+    this.validatorsLookup = {}
 
     this.dataReady = new Promise((resolve) => {
       this.resolveReady = resolve
     })
     this.dataReady.then(() => {
       console.log(`Global Store is ready`)
     })
-    this.getStores()
+    this.getReady()
   }
 
-  async getStores() {
-    const networks = await this.db.getNetworks()
-    if (networks.length !== this.stores.length) {
-      setTimeout(async () => await this.getStores(), 1000)
+  async getReady() {
+    if (Object.keys(this.store).length === 0) {
+      setTimeout(async () => await this.getReady(), 1000)
     } else {
       await this.getGlobalValidators()
       this.resolveReady()
     }
   }
 
   upsertStoreToGlobalStore(newStore) {
```

Comment:
why "upsert"? what does this do?

Reply:
it checks if the particular network store already lives in `this.stores`. If not, it will just push it in. If it does, it will update it. Hence the name.

I'll add a comment here :+1:

Reply:
but that is not what the function actually does

Reply:
OK, willl recheck it

Reply:
Debugging it looks like it is working perfectly this logic :shrug: What error do you see there?